src/com/goide/intentions/GoMoveToStructInitializationIntention.java

```diff
@@ -0,0 +1,283 @@
+                                                         @NotNull GoAssignmentStatement assignment) {
+    GoReferenceExpression selectedFieldReference = PsiTreeUtil.getTopmostParentOfType(selectedElement, GoReferenceExpression.class);
+    if (isFieldReference(selectedFieldReference)) {
+      return ObjectUtils.nullizeByCondition(selectedFieldReference, reference -> isValueAssignedInPreviousStatement(reference, assignment));
+    }
+
+    List<GoExpression> fieldReferences = ContainerUtil.filter(assignment.getLeftHandExprList().getExpressionList(),
+                                                              GoMoveToStructInitializationIntention::isFieldReference);
+    if (ContainerUtil.exists(fieldReferences, expression -> isValueAssignedInPreviousStatement(expression, assignment))) return null;
+
+    long distinctStructsCount = fieldReferences.stream()
+      .map(expression -> getQualifierName((GoReferenceExpression)expression))
+      .distinct().count();
+    return distinctStructsCount == 1 ? tryGetReference(ContainerUtil.getFirstItem(fieldReferences)) : null;
+  }
+
+  private static boolean isValueAssignedInPreviousStatement(@NotNull GoExpression expression,
+                                                            @NotNull GoAssignmentStatement assignment) {
+    GoReferenceExpression value = tryGetReference(GoPsiImplUtil.getExpressionValue(assignment, expression));
+    if (value == null) return false;
+    PsiElement previousElement = PsiTreeUtil.skipSiblingsBackward(assignment, PsiWhiteSpace.class);
+    if (previousElement instanceof GoSimpleStatement) {
+      GoShortVarDeclaration varDeclaration = ((GoSimpleStatement)previousElement).getShortVarDeclaration();
+      return varDeclaration != null &&
+             ContainerUtil.exists(varDeclaration.getVarDefinitionList(), definition -> isTextMatches(definition, value));
+    }
+    if (previousElement instanceof GoAssignmentStatement) {
+      return ContainerUtil.exists(((GoAssignmentStatement)previousElement).getLeftHandExprList().getExpressionList(),
+                                  leftExpression -> isTextMatches(leftExpression, value));
+    }
+    return false;
+  }
+
+  @NotNull
+  private static List<GoExpression> getUninitializedSingleFieldReferences(@NotNull GoAssignmentStatement assignment,
+                                                                          @NotNull GoReferenceExpression selectedFieldReference) {
+    GoCompositeLit structLiteral = getStructLiteralByReference(selectedFieldReference, assignment);
+    String qualifierName = getQualifierName(selectedFieldReference);
+    if (structLiteral == null || qualifierName == null) return ContainerUtil.emptyList();
+    List<GoExpression> uninitializedFieldsReferences =
+      ContainerUtil.filter(assignment.getLeftHandExprList().getExpressionList(), expression ->
+        isUninitializedFieldReferenceByQualifier(expression, structLiteral, qualifierName) &&
```

Comment:
you cast same expression to GoReferrenceExpression 4 times:
1 isUninitializedFieldReferenceByQualifier
2 isValueAssignedInPreviousStatement
3 ```MultiMap<String, GoExpression> fieldReferencesByName =ContainerUtil.groupBy(uninitializedFieldsReferences, expression -> getFieldName((GoReferenceExpression)expression));```
4 ``` return ContainerUtil.filter(uninitializedFieldsReferences, reference -> fieldReferencesByName.get(getFieldName((GoReferenceExpression)reference)).size() == 1);```